pkg/state/state_exec_tmpl.go

```diff
@@ -132,3 +144,76 @@ func (st *HelmState) ExecuteTemplates() (*HelmState, error) {
 
 	return &r, nil
 }
+
+type CyclicReleaseTemplateInheritanceError struct {
+	Message string
+}
+
+func (e CyclicReleaseTemplateInheritanceError) Error() string {
+	return e.Message
+}
+
+func (st *HelmState) releaseWithInheritedTemplate(r *ReleaseSpec, inheritancePath []string) (*ReleaseSpec, error) {
```

Comment:
@mumoshu could you add more comments for this function?

Reply:
Added in 1d99e63 👍